CONTRIBUTING.md

```diff
@@ -0,0 +1,89 @@
+# Contributing to lc0
+
+These are the guidelines and standards followed by this codebase.
+
+The language is C++, specifically C++14. The compiler used must support C++14, and the entire C++14 standard featureset may be used. As such, manual `new` and `delete` memory mangement is disallowed; use the standard library tools for managing memory (such as `unique_ptr`, `shared_ptr` etc.). When compiler support is more widespread, the project may upgrade to the C++17 standard in the future.
```

Comment:
Second sentence doesn't add any new information on top of the first. :)
`new` and `delete` are not disallowed, they are strongly discouraged, but sometimes necessary.

Reply:
I've heard of plenty of projects that use subsets of features from various standards, enough so that "this is C++14" may be considered ambiguous; but I don't have that much experience with C++, so I'll defer to you on this matter.